### ISSUE TITLE
How to set multiple colors in single dataset in Linechart

### ISSUE THREAD

**Issue by sanjhi** [PhilJay/MPAndroidChart#2875]
Hey PhilJay, I have a query, I have a dataset with multiple value, i.e

if (i == 0) {
                group1.add(new BarEntry(ValueOfA1, 0));
                group1.add(new BarEntry(ValueOfB1, 1));
                group1.add(new BarEntry(ValueOfC1, 2));
                group1.add(new BarEntry(ValueOfD1, 3));
            }
if (i == 1) {
                group1.add(new BarEntry(ValueOfA2, 0));
                group1.add(new BarEntry(ValueOfB2, 1));
                group1.add(new BarEntry(ValueOfC2, 2));
                group1.add(new BarEntry(ValueOfD2, 3));
            }

After that, I am inserting all the groups into linechart dataset , i.e

 if(n==0){
                linedataset = new LineDataSet(group1, "dataset1");
                linedataset.setDrawCubic(true);
                linedataset.setDrawFilled(true);
            }
            if(n==1){
                linedataset = new LineDataSet(group2, "dataset2");
                linedataset.setDrawCubic(true);
                linedataset.setDrawFilled(true);
            } 
 linedataset.setColor(getResources().getColor(colorIds[n]));
            dataSets.add(linedataset);

it generates two lines of graph that is absolutely fine, but it assign color to every line. I want to assign different color to each item or each line. I mean 
ValueOfA1 and ValueOfA2 should be in red,
ValueOfB1 and ValueOfB2 should be in green. etc etc.. 
also want to display it A B C D in legends with particular name 
how can we do that?

**Comment by Jetz72** [comment]
Have you tried setColors instead of setColor? That allows you to use an array of colors instead of just one, where colors[0] will be used for the first entry, colors[1] for the second, and so on. Sounds like you may want a custom legend instead of the automatically generated one if you go this route. There's also setCircleColors which affects the points instead of the lines connecting them.

Also not sure why you're using BarEntries with a LineDataSet when normal Entry objects do fine for line data. Not sure if that will affect the display or if it's intentional, but aside from that, using setColors is pretty much what I've been doing, and it sounds like what you're looking for.

**Comment by sanjhi** [comment]
Hi, 
Thank you for your kind reply. I have tried setColors[] as you said that colors[0] will be used for the first entry, colors[1] for the second, but the thing is, it is taking different lineDataset as a different index but not every single value inside every single lineDataset.

Also , I am using BarEntries instead of simple Entry because I don't find any difference in graph creation while using both one by one. Anyways, I have changed it as you told me.

**Comment by sanjhi** [comment]
One more thing, is there anyway to draw line between ValueOfA1 and ValueOfA2 or ValueOfB1 and ValueOfB2 so that it would be in same color always. Currently it is drawing a line with in linedataSet . I mean currently it is like this,  ValueOfA1--- ValueOfB1--- ValueOfC1, but I can we draw it like this :-
ValueOfA1---ValueOfA2---ValueOfA3.

**Comment by Jetz72** [comment]
Hmm, if you want the values colored based on A, B or C, and you want the lines connecting A1 to A2, B1 to B2, and so on, it sounds like you're looking for a structure more like this:
`linedatasetA
{
 ValueOfA1
 ValueOfA2
 ValueOfA3
}
linedatasetB
{
 ValueOfB1
 ValueOfB2
 ValueOfB3
}
linedatasetC
{
...'
And so on.

**Comment by sanjhi** [comment]
Yes, I guess this structure will work for me. But this is possible only when we have static value, but here we are having dynamic linedatasets. It could be possible that we have 2 linedatasets, or 3, or 4 up to 10(max) and every dataset has 4 values in it.

**Comment by Jetz72** [comment]
If you have the ValueOfA1, ValueOfA2, and so on stored in arrays, then it wouldn't be hard to iterate over those arrays in a loop to generate a group using the four arrays, add it into a lineDataSet, style it,, and add it to the chart's data set. If you need to setup 10 of them, it probably should be done dynamically anyway.
